fsspec/implementations/dbfs.py

```diff
@@ -219,6 +226,60 @@ def _open(self, path, mode="rb", block_size="default", **kwargs):
         """
         return DatabricksFile(self, path, mode=mode, block_size=block_size, **kwargs)
 
+    def retry_request(func, retries=5):
+
+        RETRIABLE_EXCEPTIONS = (
+            requests.exceptions.ChunkedEncodingError,
+            requests.exceptions.ConnectionError,
+            requests.exceptions.ReadTimeout,
+            requests.exceptions.Timeout,
+            requests.exceptions.ProxyError,
+            requests.exceptions.SSLError,
+            requests.exceptions.ContentDecodingError,
+
+            aiohttp.client_exceptions.ClientError,
+        )
+
+        def is_retriable(exception):
+            """Returns True if this exception is retriable."""
+            errs = list(range(500, 505)) + [
+                408,  # Request Timeout
+                429  # Too Many Requests
+            ]
+            errs += [str(e) for e in errs]
+            if type(exception) is requests.exceptions.HTTPError:
+                return exception.response.status_code in errs
+
+            return isinstance(exception, RETRIABLE_EXCEPTIONS)
+
+        def retry_wrapper(*args, **kwargs):
+            attempts = 0
+            while attempts < retries:
+                try:
+                    return func(*args, **kwargs)
+                except (
+                        DatabricksException,
+                        requests.exceptions.RequestException,
+                        requests.exceptions.HTTPError,
+                        Exception
+                ) as exc:
+                    if is_retriable(exc):
+                        if type(exc) is requests.exceptions.HTTPError:
+                            time.sleep(int(exc.response.headers["Retry-After"]))
```

Comment:
This header is not necessarily present. Furthermore, it is not necessarily an int value. In other places, it's typical to wait something like
```
base * multiplier ** attempts
```
(perhaps with a reasonable maximum wait value of 0.5s or so)